pkg/commands/build/build.go

```diff
@@ -159,12 +159,24 @@ func (o *Options) emitResources(
 	if o.outputPath != "" && fSys.IsDir(o.outputPath) {
 		return writeIndividualFiles(fSys, o.outputPath, m)
 	}
-	if o.outOrder == legacy {
+	switch o.outOrder {
+	case legacy:
 		// Done this way just to show how overall sorting
 		// can be performed by a plugin.  This particular
 		// plugin doesn't require configuration; just make
 		// it and call transform.
 		builtin.NewLegacyOrderTransformerPlugin().Transform(m)
+	case kubectlapply:
+		// The next improvment could be to read the kubectl apply
+		// order from the kustomization.yaml and pass it to the
```

Comment:
@jbrette If we put it into the bultin plugins, it can only be used in the latest kustomize. We can put it into an example plugin that can be downloaded. The the previous versions of kustomize can also use it.

Reply:
@Liujingfang1 Sorry ...I did not put much time on that PR.  I was almost ready to withdraw this PR because I could not solve the following questions:

1 On one side I understand the interested of moving it out of the builtin plugins, one the other side how does a user use something like:

```bash
kustomize build xxx | kubectl apply -f -
```
versus

```bash
kustomize build xxx | kubectl delete -f -
```

**Does he have to change the transformers list or the transformer config to adapt the sorting to the kubectl command he wants to run**. By piggy backing like I proposed in the --reorder value, this address the issue.

2. Also I was thinking that once the new kustomize is integrated into kubectl, we will have to help the integration in order to be able to detect if the user is running kubectl apply -k <xxx> or kubectl delete -k <xxx>.

3. I don't understand all the possible tweaks in kustomize yet, hence I don't understand how a user would be able to configure the "order list"  of the Transformer, without having basically copy pasting the transformer code and changing to his own list.

4. I saw multiple PR being merged which are changing that list: https://github.com/kubernetes-sigs/kustomize/blob/master/pkg/gvk/gvk.go#L83 because people are still using the 'builtin' legacyorder'.  I found a pretty difficult bug to fix in the namereferencetransformer  because of the way the fieldspec are sorted which itself relies in the gvk sorting. Some of the conflicts that should be detected or not detected because of the way we loop accross fieldspecs and apply mutatefield. Have a look at https://github.com/keleustes/kustomize/blob/residequals-deeper/pkg/transformers/namereference.go#L84. When we change the loop, the conflict is detected.
Anyway the important point here is:  **Every time we change something in the gvk.go list,...we take the risk of breaking something or change the behavior**.

In one word, I don't mind abandoning the PR or somebody taking it over. What do you think ?

Reply:
@Liujingfang1 done